`No such file or directory` when accessing non existent `ids` directory

On a new install of 0.24.0, if one runs `cargo crev trust ...` it will try to access the `ids` directory before it is created and fail with an IO error.
A new install of 0.23.0 works fine under the same constraints

The error chain:
- [fs::read_dir()](https://github.com/crev-dev/cargo-crev/blob/a3bb27c0a024ce11c8112a886f478fc7062630f1/crev-lib/src/local.rs#L281)
- [get_current_user_public_ids()](https://github.com/crev-dev/cargo-crev/blob/a3bb27c0a024ce11c8112a886f478fc7062630f1/crev-lib/src/local.rs#L1142)
- [load_db()](https://github.com/crev-dev/cargo-crev/blob/a3bb27c0a024ce11c8112a886f478fc7062630f1/cargo-crev/src/main.rs#L513)
- [run_command()](https://github.com/crev-dev/cargo-crev/blob/master/cargo-crev/src/main.rs#L1086)


If you want to report an issue, please go through the following checklist:

* If a command is failing try running it with `RUST_BACKTRACE=1`
  No new information provided
* Paste the exact command and error and/or outputs (along with the backtrace)
  ```console 
  cargo crev trust --level high --no-commit https://github.com/mobilecoinfoundation/crev-proofs
  I/O: No such file or directory (os error 2)

  Caused by:
      No such file or directory (os error 2)
  ```

* Please let us know:
    * Which version are you using (eg. `cargo crev --version`)
      0.24.0
    * How did you install `crev` (git?, cargo?, your distribution?)
      * Installing by downloading the binary
      * Building locally with `cargo run ...` or `cargo build && ./target ...`
    * What OS/platform are you running on
      * Ubuntu 22.04 (github actions)
      * Mac OS (m2)

Building 0.23 failed for me.

```
$ rustup run stable cargo install cargo-crev --version 0.23.0
...
   Compiling cargo-crev v0.23.0
error[E0063]: missing fields `distrust_distance` and `none_trust_distance` in initializer of `crev_lib::TrustDistanceParams`
   --> /home/jayvdb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-crev-0.23.0/src/opts.rs:168:9
    |
168 |         crev_lib::TrustDistanceParams {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `distrust_distance` and `none_trust_distance`

For more information about this error, try `rustc --explain E0063`.
error: could not compile `cargo-crev` (bin "cargo-crev") due to previous error
error: failed to compile `cargo-crev v0.23.0`, intermediate artifacts can be found at `/tmp/cargo-installAjpJc4`
```

--- a/crev-common/src/blake2b256.rs
+++ b/crev-common/src/blake2b256.rs
@@ -1,4 +1,4 @@
-use blake2::Blake2b;
 use blake2::digest::consts::U32;
+use blake2::Blake2b;
 
 pub type Blake2b256 = Blake2b<U32>;

--- a/crev-lib/src/local.rs
+++ b/crev-lib/src/local.rs
@@ -247,6 +247,13 @@ impl Local {
         self.user_dir_path().join("ids")
     }
 
+    /// Like [`Self::user_ids_path`] but checks if the dir exists
+    pub fn user_ids_path_opt(&self) -> Option<PathBuf> {
+        let path = self.user_dir_path().join("ids");
+
+        path.exists().then_some(path)
+    }
+
     /// Directory where git checkouts for user's own proof repos are stored
     ///
     /// This is separate from cache of other people's proofs
@@ -258,11 +265,7 @@ impl Local {
     pub fn user_proofs_path_opt(&self) -> Option<PathBuf> {
         let path = self.user_proofs_path();
 
-        if path.exists() {
-            Some(path)
-        } else {
-            None
-        }
+        path.exists().then_some(path)
     }
 
     /// Path where this Id is stored as YAML
@@ -276,13 +279,14 @@ impl Local {
 
     /// Returns public Ids which belong to the current user.
     pub fn get_current_user_public_ids(&self) -> Result<Vec<PublicId>> {
-        let ids_path = self.user_ids_path();
         let mut ids = vec![];
-        for dir_entry in std::fs::read_dir(&ids_path)? {
-            let path = dir_entry?.path();
-            if path.extension().map_or(false, |ext| ext == "yaml") {
-                let locked_id = LockedId::read_from_yaml_file(&path)?;
-                ids.push(locked_id.to_public_id())
+        if let Some(ids_path) = self.user_ids_path_opt() {
+            for dir_entry in std::fs::read_dir(&ids_path)? {
+                let path = dir_entry?.path();
+                if path.extension().map_or(false, |ext| ext == "yaml") {
+                    let locked_id = LockedId::read_from_yaml_file(&path)?;
+                    ids.push(locked_id.to_public_id())
+                }
             }
         }
 

